src/fs/rock/RockRebuild.cc

```diff
@@ -290,6 +287,9 @@ Rock::Rebuild::Start(SwapDir &dir)
     if (stats->completed(dir)) {
         debugs(47, 2, "already indexed cache_dir #" <<
                dir.index << " from " << dir.filePath);
+        // we could fail to send this before in a case of crash
+        if (UsingSmp() && IamDiskProcess())
```

Comment:
The IamDiskProcess() check feels odd here. If we are here, we are responsible for rebuilding the index. If we are in SMP mode, then we have to be a disker, right?

```suggestion
        if (UsingSmp())
```

We can assert IamDiskProcess() inside the `if` statement.